Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2332,29 +2333,37 @@ bool CSSPropertyParser::consumeGridShorthand(bool important)
 
 bool CSSPropertyParser::consumePlaceContentShorthand(bool important)
 {
+    // <'place-content'>   = <'align-content'> <'justify-content'>?
+    // <'align-content'>   = normal | <baseline-position> | <content-distribution> | <overflow-position>? [ <content-position>                ]
+    // <'justify-content'> = normal |                       <content-distribution> | <overflow-position>? [ <content-position> | left | right ]
+    // https://drafts.csswg.org/css-align/#propdef-place-content
+
+    // NOTE: The second value, if omitted, is copied from the first value.
+
     ASSERT(shorthandForProperty(CSSPropertyPlaceContent).length() == 2);
 
     if (m_range.atEnd())
         return false;
 
-    CSSParserTokenRange rangeCopy = m_range;
     bool isBaseline = isBaselineKeyword(m_range.peek().id());
-    RefPtr alignContentValue = consumeContentDistributionOverflowPosition(m_range, isContentPositionKeyword);
+
+    RefPtr alignContentValue = consumeAlignContent(m_range);
     if (!alignContentValue)
         return false;
 
-    // justify-content property does not allow the <baseline-position> values.
-    if (m_range.atEnd() && isBaseline)
-        return false;
-    if (isBaselineKeyword(m_range.peek().id()))
-        return false;
+    if (m_range.atEnd()) {
+        // justify-content property does not allow the <baseline-position> values, so if only 'align-content' is provided, it can't be a <baseline-position>.
+        if (isBaseline)
+            return false;
 
-    if (m_range.atEnd())
-        m_range = rangeCopy;
-    RefPtr justifyContentValue = consumeContentDistributionOverflowPosition(m_range, isContentPositionOrLeftOrRightKeyword);
-    if (!justifyContentValue)
-        return false;
-    if (!m_range.atEnd())
+        RefPtr justifyContentValue = alignContentValue.copyRef();
```

Comment:
Why isn't this `Ref` instead of `RefPtr`? Why is explicit `copyRef` needed?

Reply:
I don't have a good justification for the RefPtr here other than it made it so both sets of addProperty() calls were identical. As for the copyRef(), that is just a mistake, I don't think it is needed.